forte/data/ontology/core.py

```diff
@@ -865,5 +867,167 @@ def index_key(self) -> int:
+                grid, the unit is one grid cell.
+            w_idx: the zero-based width(column) index of the grid cell in the
+                grid, the unit is one grid cell.
+
+        Raises:
+            ValueError: ``h_idx`` is out of the range specified by ``height``.
+            ValueError: ``w_idx`` is out of the range specified by ``width``.
+
+        Returns:
+            numpy array that represents the grid cell.
+        """
+        if not 0 <= h_idx < self._height:
+            raise ValueError(
+                f"input parameter h_idx ({h_idx}) is"
+                "out of scope of h_idx range"
+                f" {(0, self._height)}"
+            )
+        if not 0 <= w_idx < self._width:
+            raise ValueError(
+                f"input parameter w_idx ({w_idx}) is"
+                "out of scope of w_idx range"
+                f" {(0, self._width)}"
+            )
+        # initialize a numpy zeros array
+        array = np.zeros((self.image_height, self.image_width))
+        # set grid cell entry values to the values of the original image array
+        # (entry values outside of grid cell remain zeros)
+        # An example of computing grid height index range is
+        # index * cell height : min((index + 1) * cell height, image_height).
+        # It's similar for computing cell width index range
+        # Plus, we constrain the maximum pixel locations by the image size as
+        # the last grid cell per row and column might be out of the image size
+        array[
+            h_idx * self.c_h : min((h_idx + 1) * self.c_h, self.image_height),
+            w_idx * self.c_w : min((w_idx + 1) * self.c_w, self.image_width),
+        ] = img_arr[
+            h_idx * self.c_h : min((h_idx + 1) * self.c_h, self.image_height),
+            w_idx * self.c_w : min((w_idx + 1) * self.c_w, self.image_width),
+        ]
+        return array
+
+    def get_grid_cell_center(self, h_idx: int, w_idx: int) -> Tuple[int, int]:
+        """
+        Get the center position of the grid cell in the ``Grid``.
+
+        Note: all indices are zero-based and counted from top left corner of
+        the grid.
+
+        Args:
+            h_idx: the height(row) index of the grid cell in the grid,
+                , the unit is one image array entry.
+            w_idx: the width(column) index of the grid cell in the
+                grid, the unit is one image array entry.
+
+        Returns:
+            A tuple of (y index, x index)
+        """
+        # compute the center position of the grid cell by
+        # dividing the grid cell height range and wdith range by 2 (round down)
+        # suppose an extreme case that
+        # a grid cell has
+        # a height range of (0, 3) and a width range of (0, 3)
+        # the grid cell center would be (1, 1)
+        # since the grid cell size is usually very large
+        # the minor offset of the grid cell center usually doesn't matter
```

Comment:
Behavior-related documentation needs to be in the docstring, not inline comments.